### PR TITLE
feat!: added datadog loggin service and made changes accordingly with respective readme update

### PR DESCRIPTION
**Pull request by abdullahwaheed**
BREAKING CHANGE: removed NewRelicLoggingService and introduced DatadogLoggingService and updated repo structure

### CODE REVIEW

**Review by schenedx — APPROVED** [review]
👍

File: package.json [hunk 1]
```diff
@@ -1,23 +1,15 @@
...
-    "commitmsg": "commitlint -e $GIT_PARAMS",
-    "coveralls": "cat ./coverage/lcov.info | coveralls",
-    "lint": "eslint --ext .js --ext .jsx .",
-    "is-es5": "es-check es5 ./dist/*.js",
-    "precommit": "npm run lint",
-    "prepublishOnly": "npm run build",
-    "semantic-release": "semantic-release",
-    "test": "jest --coverage",
-    "travis-deploy-once": "travis-deploy-once"
+    "build": "./node_modules/.bin/fedx-scripts babel src --out-dir dist --source-maps --copy-files",
```
Comment (adamstankiewicz) [inline package.json:10]: The `dist` directory contains more files than should be published. For example, the `setupTest.js` file nor any test files should be published, as it's not relevant to consumers' use of the package. The `dist` build output is also missing the `LICENSE`, `README`, and `package.json`.

image

See the [previously linked `build` Makefile target](https://github.com/openedx/frontend-platform/blob/05e6820f627ac0d3e4dcf68858191f2a5b8e42c6/Makefile#L13-L21) in frontend-platform for reference on what the `build` NPM script would ideally be doing here.

```
"build": "make build",
```

Comment (abdullahwaheed) [reply package.json:10]: should we add makefile here as well?

Comment (adamstankiewicz) [reply package.json:10]: Yes. Given there's multiple commands to `rm` and `cp`, the `Makefile`s `build` target is preferable over having the copying/removal of files all in one line within `package.json`.

File: README.md [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+    const config = {
+      loggingService: DatadogLoggingService,
+    };
+    
+    export default config;
+
+## DatadogLoggingService
+
+The DatadogLoggingService is a concrete implementation of the logging service interface that sends messages to Datadog that can be seen in Datadog RUM and Datadog browser logs. When in development mode, all messages will instead be sent to the console.
```
[the remark this answers is not in the document]
Comment (abdullahwaheed) [reply README.md:32]: added in the end of file

File: package.json [hunk 3]
```diff
@@ -1,23 +1,15 @@
 {
   "name": "@edx/frontend-logging",
   "version": "1.0.0-semantically-released",
-  "description": "Frontend logging client code for sending logging and events to New Relic",
-  "main": "dist/main.js",
+  "description": "Edx Frontend logging client code for sending logging and events to Datadog",
+  "main": "dist/index.js",
   "publishConfig": {
```
[the remark this answers is not in the document]
Comment (abdullahwaheed) [reply package.json:7]: added sideEffect: false

File: .github/ci.yml [hunk 4]
```diff
@@ -0,0 +1,37 @@
+name: node_js CI
+
+on:
+  push:
+    branches:
+      - master
```
Comment (adamstankiewicz) [inline .github/ci.yml:6]: nit: From the [previous comment](https://github.com/edx/frontend-logging/pull/20#discussion_r1594034081), I would still recommend only running this for `pull_request`. There's not really a need to run this workflow for commits to `master`.

**Review by adamstankiewicz — APPROVED** [review]

**Pull request merged**